src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs

```diff
@@ -143,11 +149,116 @@ public ObservableCollection<string> NumberFormatList
+            get
+            {
+                return _pythonEngineList;
+            }
+            set
+            {
+                _pythonEngineList = value;
+                RaisePropertyChanged("PythonEnginesList");
+            }
+        }
+
+        /// <summary>
+        /// Controls the Selected option in Number Format ComboBox
+        /// </summary>
+        public string SelectedPythonEngine
+        {
+            get
+            {
+                return selectedPythonEngine;
+            }
+            set
+            {
+                selectedPythonEngine = value;
+                RaisePropertyChanged("SelectedPythonEngine");
+            }
+        }
+        
+        /// <summary>
+        /// Controls the IsChecked property in the "Hide IronPython alerts" toogle button
+        /// </summary>
+        public bool HideIronPythonAlertsIsChecked
+        {
+            get
+            {
+                return hideIronPAlerts;
+            }
+            set
+            {
+                hideIronPAlerts = value;
+                RaisePropertyChanged("HideIronPythonAlerts");
+            }
+        }
+
+        /// <summary>
+        /// Controls the IsChecked property in the "Show Whitespace in Python editor" toogle button
+        /// </summary>
+        public bool ShowWhitespaceIsChecked
+        {
+            get
+            {
+                return showWhitespace;
+            }
+            set
+            {
+                showWhitespace = value;
+                RaisePropertyChanged("ShowWhitespace");
+            }
+        }
+
+        /// <summary>
+        /// Controls the IsChecked property in the "Node autocomplete" toogle button
+        /// </summary>
+        public bool NodeAutocompleteIsChecked
+        {
+            get
+            {
+                return nodeAutocomplete;
+            }
+            set
+            {
+                nodeAutocomplete = value;
+                RaisePropertyChanged("NodeAutocomplete");
+            }
+        }
+
+        /// <summary>
+        /// Controls the IsChecked property in the "Enable T-spline nodes" toogle button
+        /// </summary>
+        public bool EnableTSplineIsChecked
+        {
+            get
+            {
+                return enableTSpline;
+            }
+            set
+            {
+                enableTSpline = value;
+                RaisePropertyChanged("EnableTSpline");
+            }
+        }
+        #endregion
+
         /// <summary>
         /// The PreferencesViewModel constructor basically initialize all the ItemsSource for the corresponding ComboBox in the View (PreferencesView.xaml)
         /// </summary>
         public PreferencesViewModel()
         {
+            PythonEnginesList = new ObservableCollection<string>();
+            PythonEnginesList.Add(Wpf.Properties.Resources.DefaultPythonEngineNone);
+            PythonEnginesList.Add("IronPython2");
+            PythonEnginesList.Add("CPython3");
```

Comment:
I dont really like this but seems there is not an easy way to get the enums defined in PythonNodeModels.dll without adding the reference.. Which is not ideal either..

Reply:
That is correct, I had issues adding that reference

Reply:
one way is to look for the enum type at runtime, if the lookup fails, then you can do this. The only benefit to that approach is to have the menu items update automatically - both of them are brittle in their own ways.

Reply:
@mjkkirschner I see, I think it is worth a try, especially with the new preferences panel, when user access it, node libs should be loaded 100%. @Astul-Betizagasti can you look at the code example provided by Mike and give it a try? There maybe other ways, like https://stackoverflow.com/questions/5005931/listing-values-within-enums-using-reflection-in-c-sharp which also worth a look

Reply:
Just added a commit that should cover this changes, let me know if you have any comments, thanks!
@QilongTang @mjkkirschner